internal/controller/postgrescluster/pgmonitor.go

```diff
@@ -221,25 +221,38 @@ func (r *Reconciler) reconcileMonitoringSecret(
 
 	intent.Data = make(map[string][]byte)
 
-	if len(existing.Data["password"]) == 0 || len(existing.Data["verifier"]) == 0 {
+	// Copy existing password and verifier into the intent
+	password, set := existing.Data["password"]
+	if set {
```

Comment:
🤔 Why distinguish between unset and zero/empty here when the blocks below don't?

(Similar code exists [over here](https://github.com/CrunchyData/postgres-operator/blob/eedf513582b7d90c7763f64eb55c5bb266d729f5/internal/controller/postgrescluster/postgres.go#L69-L73).)

Reply:
Oh yeah, we could just set the value from existing (assuming it isn't nil) every time and operator based on the intent value going forward. I'll look at changing this